[Cloud Security] Adds KQL filter to graph component

kfirpeled opened the pull request:
## Summary

This PR includes the following changes to the graph investigation component:
- Added KQL filter to graph investigation component
- Shows toast message on syntax error
- Includes the KQL filter in timeline investigation through the graph

https://github.com/user-attachments/assets/4653bdfd-277a-4479-abc2-65a90fd50f57

**How to test:**

To test this PR using storybook (alternatively access to storybooks attached to this build)

```
yarn storybook cloud_security_posture_packages
```

To test e2e:

- Enable the feature flag 

`kibana.dev.yml`:

```yaml
uiSettings.overrides.securitySolution:enableVisualizationsInFlyout: true
xpack.securitySolution.enableExperimental: ['graphVisualizationInFlyoutEnabled']
```

- Load mocked data:

```bash
node scripts/es_archiver load x-pack/test/cloud_security_posture_functional/es_archives/logs_gcp_audit \ 
  --es-url http://elastic:changeme@localhost:9200 \
  --kibana-url http://elastic:changeme@localhost:5601

node scripts/es_archiver load x-pack/test/cloud_security_posture_functional/es_archives/security_alerts \
  --es-url http://elastic:changeme@localhost:9200 \
  --kibana-url http://elastic:changeme@localhost:5601
```

- Make sure you include data from Oct 13 2024. (in the video I use Last year)

To run FTR tests:

```
yarn test:ftr:server --config x-pack/test/cloud_security_posture_functional/config.ts
yarn test:ftr:runner --config x-pack/test/cloud_security_posture_functional/config.ts --grep="Graph visualization"
```

<details>
<summary>E2E tests 📹 </summary>


https://github.com/user-attachments/assets/5f00d911-fc8f-4329-bb1d-0468a260fef2



</details>

### Checklist

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed


<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!

kfirpeled committed:
Added kql filter to graph investigation
- Filters based on KQL filter
- Added e2e tests and UT
- Shows toast message on syntax error
- Adapts the kql filter when open timeline for investigation

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/265201)
* Commit: 571da797778055b09238362d23d22d40fa7a2072

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/265201#01944663-9637-4d87-9dab-ae872143e6b1)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265201","number":265201,"commit":"571da797778055b09238362d23d22d40fa7a2072"}],"number":265201}
buildkite-pr-comment-->

kfirpeled committed:
fix i18n key

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/265214)
* Commit: edce90df70c562ca8d1c7c1e0c1628a229c78ec1

### Failed CI Steps
* [Serverless Entity Analytics - Security Cypress Tests #3](https://buildkite.com/elastic/kibana-pull-request/builds/265214#01944675-852a-4e77-bae8-6c674cbc8be9)

### History
* :broken_heart: [Build #265201](https://buildkite.com/elastic/kibana-pull-request/builds/265201) failed 571da797778055b09238362d23d22d40fa7a2072

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265214","number":265214,"commit":"edce90df70c562ca8d1c7c1e0c1628a229c78ec1"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265201","number":265201,"commit":"571da797778055b09238362d23d22d40fa7a2072"}],"number":265214}
buildkite-pr-comment-->

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#7669](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/7669)
[✅] x-pack/test/cloud_security_posture_functional/config.ts: 25/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F205570%2Fhead)

kfirpeled committed:
Merge branch 'main' into cspm/graph-add-kql-filter

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/265341)
* Commit: d5e4b770b15b570a927bc5f4cd8c647d22a0d892

### Failed CI Steps
* [x-pack/test_serverless/functional/test_suites/observability/common_configs/config.group5.ts](https://buildkite.com/elastic/kibana-pull-request/builds/265341#01944759-b06d-4dfb-8bd4-c3dd2fab9082)
* [x-pack/test_serverless/functional/test_suites/observability/common_configs/config.group5.ts](https://buildkite.com/elastic/kibana-pull-request/builds/265341#0194475f-4f04-4194-8494-7e46625a6a6d)
* [x-pack/test_serverless/functional/test_suites/security/common_configs/config.group5.ts](https://buildkite.com/elastic/kibana-pull-request/builds/265341#01944759-b06e-40b2-93d9-ea362658f13a)
* [x-pack/test_serverless/functional/test_suites/security/common_configs/config.group6.ts](https://buildkite.com/elastic/kibana-pull-request/builds/265341#01944759-b06f-4f89-bb26-c3f1079d9fd9)
* [FTR Configs #13](https://buildkite.com/elastic/kibana-pull-request/builds/265341#01944759-b07a-4d59-a33f-3b2b4b90d96c)
* [FTR Configs #33](https://buildkite.com/elastic/kibana-pull-request/builds/265341#01944759-b08b-44a3-b7f8-54a70c09cb72)
* [FTR Configs #53](https://buildkite.com/elastic/kibana-pull-request/builds/265341#01944759-b09b-4acb-a301-38a0d9ee3e6d)
* [FTR Configs #68](https://buildkite.com/elastic/kibana-pull-request/builds/265341#01944759-b0a5-487d-bd82-5751f7d1ba09)
* [FTR Configs #71](https://buildkite.com/elastic/kibana-pull-request/builds/265341#01944759-b0a6-4102-825a-3484d5e2e0eb)
* [FTR Configs #76](https://buildkite.com/elastic/kibana-pull-request/builds/265341#01944759-b0aa-45fd-ad92-b375577c21a7)
* [FTR Configs #78](https://buildkite.com/elastic/kibana-pull-request/builds/265341#01944759-b0ac-4868-8182-05569718e3e6)

### History
* :broken_heart: [Build #265214](https://buildkite.com/elastic/kibana-pull-request/builds/265214) failed edce90df70c562ca8d1c7c1e0c1628a229c78ec1
* :broken_heart: [Build #265201](https://buildkite.com/elastic/kibana-pull-request/builds/265201) failed 571da797778055b09238362d23d22d40fa7a2072

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265341","number":265341,"commit":"d5e4b770b15b570a927bc5f4cd8c647d22a0d892"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265214","number":265214,"commit":"edce90df70c562ca8d1c7c1e0c1628a229c78ec1"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265201","number":265201,"commit":"571da797778055b09238362d23d22d40fa7a2072"}],"number":265341}
buildkite-pr-comment-->

kfirpeled committed:
refactoring

kfirpeled committed:
Merge branch 'main' into cspm/graph-add-kql-filter

opauloh commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/graph_investigation/graph_investigation.stories.test.tsx:
```diff
@@ -51,6 +77,31 @@ describe('GraphInvestigation Component', () => {
+  });
+
+  it('calls refresh on submit button click', () => {
+    const mockRefresh = action('refresh');
```
For better maintainability, we should define a central place where we store Storybook actions in a constant, otherwise, as it grows it can become tricky to understand where actions such as `refresh` and `notifications:addError` are coming from

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/265402)
* Commit: e42dbbe0498ad495d048465af8de58a78d5e8dd2

### Failed CI Steps
* [x-pack/test_serverless/functional/test_suites/search/common_configs/config.group5.ts](https://buildkite.com/elastic/kibana-pull-request/builds/265402#01944809-37c9-42c0-8ff2-47d854cb7a85)
* [x-pack/test_serverless/functional/test_suites/search/common_configs/config.group5.ts](https://buildkite.com/elastic/kibana-pull-request/builds/265402#01944810-46e7-4ce7-bc8b-2505e613e670)
* [x-pack/test_serverless/functional/test_suites/search/common_configs/config.group6.ts](https://buildkite.com/elastic/kibana-pull-request/builds/265402#01944809-37c9-4034-ad53-e98663a193f0)
* [FTR Configs #58](https://buildkite.com/elastic/kibana-pull-request/builds/265402#01944809-37ec-4270-b471-85a414135152)
* [FTR Configs #61](https://buildkite.com/elastic/kibana-pull-request/builds/265402#01944809-37ed-48b8-8c2c-0f0110ed9746)
* [FTR Configs #62](https://buildkite.com/elastic/kibana-pull-request/builds/265402#01944809-37ee-4bad-bd30-30e085e036b1)
* [FTR Configs #63](https://buildkite.com/elastic/kibana-pull-request/builds/265402#01944809-37ef-4279-b6c0-f4fef6974e9d)
* [FTR Configs #64](https://buildkite.com/elastic/kibana-pull-request/builds/265402#01944809-37ef-4448-be29-60a422b86c11)
* [FTR Configs #91](https://buildkite.com/elastic/kibana-pull-request/builds/265402#01944809-3804-4bc9-8bd8-3866cc536f87)

### History
* :broken_heart: [Build #265341](https://buildkite.com/elastic/kibana-pull-request/builds/265341) failed d5e4b770b15b570a927bc5f4cd8c647d22a0d892
* :broken_heart: [Build #265214](https://buildkite.com/elastic/kibana-pull-request/builds/265214) failed edce90df70c562ca8d1c7c1e0c1628a229c78ec1
* :broken_heart: [Build #265201](https://buildkite.com/elastic/kibana-pull-request/builds/265201) failed 571da797778055b09238362d23d22d40fa7a2072

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265402","number":265402,"commit":"e42dbbe0498ad495d048465af8de58a78d5e8dd2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265341","number":265341,"commit":"d5e4b770b15b570a927bc5f4cd8c647d22a0d892"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265214","number":265214,"commit":"edce90df70c562ca8d1c7c1e0c1628a229c78ec1"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265201","number":265201,"commit":"571da797778055b09238362d23d22d40fa7a2072"}],"number":265402}
buildkite-pr-comment-->

kfirpeled committed:
fix e2e tests

albertoblaz reviewed (approved):
Haven't tested it, but code looks good! 👏

kfirpeled committed:
Merge branch 'main' into cspm/graph-add-kql-filter

kfirpeled committed:
refactor actions

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/265597)
* Commit: 8e2dd1a71dd4daa618f38a08d4bca52418d7350b
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-205570/8e2dd1a71dd4daa618f38a08d4bca52418d7350b)

### Failed CI Steps
* [Jest Tests #2](https://buildkite.com/elastic/kibana-pull-request/builds/265597#01944b4c-3d79-4bc8-a815-ee240995e217)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/265597#01944b4c-3d79-4bc8-a815-ee240995e217) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/265597/jobs/01944b4c-3d79-4bc8-a815-ee240995e217/artifacts/01944b7c-6a8b-43e6-ad96-290ad8228b51) Jest Tests #<span></span>2 / CasesWebhookActionConnectorFields renders Validation validates correctly "webhookCreateUrlText"

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/0c0dd631bbd22859a8d8e4214b923c2afafc0f11) | [after](https://github.com/elastic/kibana/commit/8e2dd1a71dd4daa618f38a08d4bca52418d7350b) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 22.2MB | 22.2MB | +1.5KB |



### History
* :broken_heart: [Build #265402](https://buildkite.com/elastic/kibana-pull-request/builds/265402) failed e42dbbe0498ad495d048465af8de58a78d5e8dd2
* :broken_heart: [Build #265341](https://buildkite.com/elastic/kibana-pull-request/builds/265341) failed d5e4b770b15b570a927bc5f4cd8c647d22a0d892
* :broken_heart: [Build #265214](https://buildkite.com/elastic/kibana-pull-request/builds/265214) failed edce90df70c562ca8d1c7c1e0c1628a229c78ec1
* :broken_heart: [Build #265201](https://buildkite.com/elastic/kibana-pull-request/builds/265201) failed 571da797778055b09238362d23d22d40fa7a2072

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265597","number":265597,"commit":"8e2dd1a71dd4daa618f38a08d4bca52418d7350b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265402","number":265402,"commit":"e42dbbe0498ad495d048465af8de58a78d5e8dd2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265341","number":265341,"commit":"d5e4b770b15b570a927bc5f4cd8c647d22a0d892"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265214","number":265214,"commit":"edce90df70c562ca8d1c7c1e0c1628a229c78ec1"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265201","number":265201,"commit":"571da797778055b09238362d23d22d40fa7a2072"}],"number":265597}
buildkite-pr-comment-->

opauloh commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/.storybook/constants.ts:
```diff
@@ -11,3 +11,21 @@ export const TITLE = 'Cloud Security Posture Storybook';
 export const URL =
   'https://github.com/elastic/kibana/tree/main/x-pack/packages/kbn-cloud-security-posture';
+
+export const WEB_STORAGE_CLEAR_ACTION = 'web_storage:clear' as const;
+export const WEB_STORAGE_GET_ITEM_ACTION = 'web_storage:getItem' as const;
+export const WEB_STORAGE_KEY_ACTION = 'web_storage:key' as const;
+export const WEB_STORAGE_REMOVE_ITEM_ACTION = 'web_storage:removeItem' as const;
+export const WEB_STORAGE_SET_ITEM_ACTION = 'web_storage:setItem' as const;
+export const STORAGE_SET_ACTION = 'storage:set' as const;
+export const STORAGE_REMOVE_ACTION = 'storage:remove' as const;
+export const STORAGE_CLEAR_ACTION = 'storage:clear' as const;
+export const NOTIFICATIONS_SHOW_ACTION = 'notifications:show' as const;
+export const NOTIFICATIONS_SUCCESS_ACTION = 'notifications:success' as const;
+export const NOTIFICATIONS_WARNING_ACTION = 'notifications:warning' as const;
+export const NOTIFICATIONS_DANGER_ACTION = 'notifications:danger' as const;
+export const NOTIFICATIONS_ADD_ERROR_ACTION = 'notifications:addError' as const;
+export const NOTIFICATIONS_ADD_SUCCESS_ACTION = 'notifications:addSuccess' as const;
+export const NOTIFICATIONS_ADD_WARNING_ACTION = 'notifications:addWarning' as const;
+export const NOTIFICATIONS_REMOVE_ACTION = 'notifications:remove' as const;
+export const EDIT_DATA_VIEW_ACTION = 'editDataView' as const;
```
💯

opauloh reviewed (approved):
LGTM

kfirpeled committed:
Merge branch 'main' into cspm/graph-add-kql-filter

PhilippeOberti reviewed (approved):
only additive optional changes that don't impact the rest of the components using the `use_investigate_in_timeline` hook. LGTM for the Threat Hunting Investigations team!

kfirpeled committed:
Merge branch 'main' into cspm/graph-add-kql-filter

kibanamachine commented:
Starting backport for target branches: 8.x

https://github.com/elastic/kibana/actions/runs/12759733373

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/266769)
* Commit: 97045e3ee6d9d8b9e045eb8e01d434a3333d5bd9
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-205570/97045e3ee6d9d8b9e045eb8e01d434a3333d5bd9)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/317a8979765ec88fc71a6faf1570462545308ee9) | [after](https://github.com/elastic/kibana/commit/97045e3ee6d9d8b9e045eb8e01d434a3333d5bd9) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 21.2MB | 21.2MB | +1.5KB |



### History
* :yellow_heart: [Build #265597](https://buildkite.com/elastic/kibana-pull-request/builds/265597) was flaky 8e2dd1a71dd4daa618f38a08d4bca52418d7350b
* :broken_heart: [Build #265402](https://buildkite.com/elastic/kibana-pull-request/builds/265402) failed e42dbbe0498ad495d048465af8de58a78d5e8dd2
* :broken_heart: [Build #265341](https://buildkite.com/elastic/kibana-pull-request/builds/265341) failed d5e4b770b15b570a927bc5f4cd8c647d22a0d892
* :broken_heart: [Build #265214](https://buildkite.com/elastic/kibana-pull-request/builds/265214) failed edce90df70c562ca8d1c7c1e0c1628a229c78ec1
* :broken_heart: [Build #265201](https://buildkite.com/elastic/kibana-pull-request/builds/265201) failed 571da797778055b09238362d23d22d40fa7a2072

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/266769","number":266769,"commit":"97045e3ee6d9d8b9e045eb8e01d434a3333d5bd9"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265597","number":265597,"commit":"8e2dd1a71dd4daa618f38a08d4bca52418d7350b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265402","number":265402,"commit":"e42dbbe0498ad495d048465af8de58a78d5e8dd2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265341","number":265341,"commit":"d5e4b770b15b570a927bc5f4cd8c647d22a0d892"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265214","number":265214,"commit":"edce90df70c562ca8d1c7c1e0c1628a229c78ec1"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265201","number":265201,"commit":"571da797778055b09238362d23d22d40fa7a2072"}],"number":266769}
buildkite-pr-comment-->

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/206534">](https://github.com/elastic/kibana/pull/206534)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->
